swift-sdk/Internal/ApiClient.swift

```diff
@@ -126,20 +105,27 @@ class ApiClient {
     }
     
     private func createIterableHeaders() -> [String: String] {
-        [JsonKey.contentType.jsonKey: JsonValue.applicationJson.jsonStringValue,
-         JsonKey.Header.sdkPlatform: JsonValue.iOS.jsonStringValue,
-         JsonKey.Header.sdkVersion: IterableAPI.sdkVersion,
-         JsonKey.Header.apiKey: apiKey]
+        var headers = [JsonKey.contentType.jsonKey: JsonValue.applicationJson.jsonStringValue,
+                       JsonKey.Header.sdkPlatform: JsonValue.iOS.jsonStringValue,
+                       JsonKey.Header.sdkVersion: IterableAPI.sdkVersion,
+                       JsonKey.Header.apiKey: apiKey]
+        
+        if let authToken = authProvider?.auth.authToken {
+            headers[JsonKey.Header.authorization] = "Bearer \(authToken)"
+        }
+        
+        return headers
     }
     
     private let apiKey: String
-    private weak var authProvider: AuthProvider?
```

Comment:
Why do we need to have a strong reference to authProvider here.

Reply:
ApiClient has a reference to AuthProvider (APIInternal) and AuthProvider (ApiInternal) has a reference to ApiClient.

Reply:
When I was writing the test, it wouldn't succeed without being strongly referenced; somewhere along the way it kept getting lost. I'll take any suggestions on how to fix this while keeping it weakly referenced, but I couldn't think of one at the time.

Reply:
Resolved, problem was in test implementation. Other tests to be fixed at a later date.